docs/book/v2/coding-style-guide.md

```diff
@@ -1,192 +1,822 @@
-- If a function throws any exceptions, they should be documented in `@throws` tags.
-  [*](ruleset.md#squizcommentingfunctioncomment)
-- The `@var` tag may be used in inline comments to document the _Type_ of properties. [*](ruleset.md#slevomatcodingstandardcommentinginlinedoccommentdeclaration)
-- Single-line comments with a `@var` tag should be written as one-liners. [*](ruleset.md#slevomatcodingstandardcommentingrequireonelinepropertydoccomment)
-- Shorthand scalar typehint variants must be used in docblocks. [*](ruleset.md#slevomatcodingstandardtypehintslongtypehints)
-
-## Declare Statements, Namespace, and Import Statements
-
-In addition to [PSR-12](https://github.com/php-fig/fig-standards/blob/master/proposed/extended-coding-style-guide.md#3-declare-statements-namespace-and-import-statements):
-
-- Each PHP file should have a page level docblock with `@see`, `@copyright` and `@license`. The copyright date should
-  only be adjusted if the file has changed.
-- Each PHP file should have a strict type declaration at the top after the page level docblock. [*](ruleset.md#slevomatcodingstandardtypehintsdeclarestricttypes)
-- Import statements should be alphabetically sorted. [*](ruleset.md#webimpresscodingstandardphpinstantiatingparenthesis)
-- Import statements should not be grouped. [*](ruleset.md#slevomatcodingstandardnamespacesdisallowgroupuse)
-- Each import statement should be on its own line. [*](ruleset.md#slevomatcodingstandardnamespacesmultipleusesperline)
-- Absolute class name references, functions and constants should be imported. [*](ruleset.md#slevomatcodingstandardnamespacesreferenceusednamesonly)
-- Unused import statements are not allowed. [*](ruleset.md#webimpresscodingstandardnamespacesunusedusestatement)
-- Classes and function within the same namespace should not be imported. [*](ruleset.md#webimpresscodingstandardnamespacesunusedusestatement)
-- Imports should not have an alias with the same name. [*](ruleset.md#slevomatcodingstandardnamespacesuselessalias)
-- A class should not have unused private constants, (or write-only) properties and methods. [*](ruleset.md#slevomatcodingstandardclassesunusedprivateelements)
+use const Vendor\Package\{ConstantA, ConstantB, ConstantC};
 
-```php
+class Foo extends Bar implements FooInterface
+{
+    public function sampleFunction(int $a, int $b = null): array
+    {
+        if ($a === $b) {
+            bar();
+        } elseif ($a > $b) {
+            $foo->bar($arg1);
+        } else {
+            BazClass::bar($arg2, $arg3);
+        }
+    }
+
+    final public static function bar()
+    {
+        // method body
+    }
+}
+~~~
+
+## 2. General
+
+### 2.1 Basic Coding Standard
+
+Code MUST follow all rules outlined in [PSR-1].
+
+The term 'StudlyCaps' in PSR-1 MUST be interpreted as PascalCase where the first letter of
+each word is capitalized including the very first letter.
+
+> **Additional ZendFramework rules**
```

Comment:
For hints in blockquotes please use always a headline with the third order. (`> ### …`)

The same here: the name of the framework is written with space.

Reply:
I don't understand this? Are you using this to indicate a sub chapter? Because these are not sub chapters. They are just additions to to chapters in random places as you can see in chapter 4.2 and 4.3.

How would this work if it is an addition to chapter 1.2.3? Should I use `> #### ...`?

Reply:
This has nothing to do with the headline order (outline) of the entire document. We use this for all first headlines in blockquotes.

Reply:
Fixed. Thanx @froschdesign for explaining.